lib/health_check.rb

```diff
@@ -0,0 +1,91 @@
+class HealthCheck
+  include ChefResource
+
+  REACHABLE = 'reachable'.freeze
+  TIMEOUT = 'timeout'.freeze
+  UNREACHABLE = 'unreachable'.freeze
+  ERRORING = 'erroring'.freeze
+  KEYERROR = 'key error'.freeze
+
+  attr_reader :status, :erchef, :postgres
+
+  def initialize
+    @status = nil
+    @erchef = { status: REACHABLE }
+    @postgres = { status: REACHABLE }
+  end
+
+  #
+  # Do a general health check.
+  #
+  def check
+    erchef_health
+    erchef_authentication
+    postgres_health
+    overall
+  end
+
+  private
+
+  #
+  # Try to ping erchef as an indicator of general health
+  #
+  def erchef_health
+    erchef_health_metric do
+      chef.get_rest 'license'
+    end
+  end
+
+  #
+  # Try to authenticate with erchef to prove that oc-id is actually functioning
+  #
+  # TODO: in order for this to work, we need a test user, with
+  # username/password stored in the vault and accessible in Settings
+  #
+  def erchef_authentication
+  end
+
+  #
+  # Check the number of active Postgres connections as an indicator of
+  # general health
+  #
+  def postgres_health
+    begin
+      ActiveRecord::Base.connection.query(
+        'SELECT count(*) FROM pg_stat_activity'
+      ).flatten.first.to_i.tap do |connections|
```

Comment:
oc-id only runs on ruby 1.9+ correct? Just double checking, since tap is a 1.9+ method.